syft/frameworks/torch/he/fv/key_generator.py

```diff
@@ -32,15 +39,64 @@ def keygen(self):
         self._generate_pk()
         return [self._secret_key, self._public_key]
 
-    def _generate_sk(self, is_initialized=False):
+    def _generate_sk(self):
         param = self._context.param
 
-        if not is_initialized:
-            self._secret_key = SecretKey(sample_poly_ternary(param))
+        self._secret_key = SecretKey(sample_poly_ternary(param))
+        self._secret_key_array = [self._secret_key]
 
     def _generate_pk(self):
         if self._secret_key is None:
             raise RuntimeError("cannot generate public key for unspecified secret key")
 
         public_key = encrypt_symmetric(self._context, self._secret_key.data)
         self._public_key = PublicKey(public_key.data)
+
+    def get_relin_keys(self):
```

Comment:
Maybe it's worth noting SEAL's design here:
 - Having a root class KSwitchKey https://github.com/microsoft/SEAL/blob/master/native/src/seal/kswitchkeys.h
 - Deriving the RelinKeys and GaloisKeys classes from that https://github.com/microsoft/SEAL/blob/master/native/src/seal/relinkeys.h


Maybe it's cleaner to have the generation logic directly in the RelinKeys class here too